test/unit_tests/backup/BackupRepositorySpec.js

```diff
@@ -52,4 +68,186 @@ describe('z.backup.BackupRepository', () => {
       expect(metaDescription.version).toBe(TestFactory.backup_service.getDatabaseVersion());
     });
   });
+
+  describe('"generateHistory"', () => {
+    const eventStoreName = z.storage.StorageSchemata.OBJECT_STORE.EVENTS;
+
+    beforeEach(() => {
+      return Promise.all([
+        ...messages.map(message => TestFactory.storage_service.save(eventStoreName, undefined, message)),
+        TestFactory.storage_service.save('conversations', conversationId, conversation),
+      ]);
+    });
+
+    afterEach(() => {
+      return TestFactory.storage_service.clearStores();
+    });
+
+    it('generate an archive of the database', () => {
+      const backupRepository = new z.backup.BackupRepository(
+        TestFactory.backup_service,
+        TestFactory.client_repository,
+        TestFactory.user_repository
+      );
+
+      const tables = z.backup.BackupService.CONFIG.SUPPORTED_TABLES;
+
+      const archivePromise = backupRepository.generateHistory(noop);
+
+      return archivePromise.then(zip => {
+        const fileNames = Object.keys(zip.files);
+        expect(fileNames).toContain('export.json');
+        tables.map(table => expect(fileNames).toContain(`${table}.json`));
+
+        const validateConversationsPromise = zip.files['conversations.json']
+          .async('string')
+          .then(conversationsStr => JSON.parse(conversationsStr))
+          .then(conversations => expect(conversations).toEqual([conversation]));
+
+        const validateEventsPromise = zip.files['events.json']
+          .async('string')
+          .then(eventsStr => JSON.parse(eventsStr))
+          .then(events => expect(events).toEqual(messages));
+
+        return Promise.all([validateConversationsPromise, validateEventsPromise]);
+      });
+    });
+
+    it('ignore verification events in the backup', () => {
+      const backupRepository = new z.backup.BackupRepository(
+        TestFactory.backup_service,
+        TestFactory.client_repository,
+        TestFactory.user_repository
+      );
+
+      const verificationEvent = {
+        conversation: conversationId,
+        type: z.event.Client.CONVERSATION.VERIFICATION,
+      };
+
+      return TestFactory.storage_service
+        .save(z.storage.StorageSchemata.OBJECT_STORE.EVENTS, undefined, verificationEvent)
+        .then(() => {
+          const archivePromise = backupRepository.generateHistory(noop);
+
+          return archivePromise.then(zip => {
+            return zip.files[z.storage.StorageSchemata.OBJECT_STORE.EVENTS]
+              .async('string')
+              .then(eventsStr => JSON.parse(eventsStr))
+              .then(events => {
+                expect(events).not.toContain(verificationEvent);
+                expect(events).toHaveLength(messages);
+              });
+          });
+        })
+        .catch(console.error.bind(console));
```

Comment:
What the reason for using `console.error.bind(console)`?

Reply:
Humm good question. Probably for debugging purposes. I can remove it now!